src/particles.cpp

```diff
@@ -316,11 +320,12 @@ void ParticleSpawner::step(float dtime, ClientEnvironment* env)
 			{
 				m_amount--;
 
-				// Pretend to, but don't actually spawn a
-				// particle if it is attached to an unloaded
-				// object.
-				if (!unloaded) {
```

Comment:
as there are more useless calculs when !unloaded, please preserve !unloaded as a parent condition and instance your variables inside it